File: app/src/main/java/com/ht117/demo/WelcomeActivity.java
package com.ht117.demo;

import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.SeekBar;
import android.widget.TextView;

import com.ht117.htui.Utils;
import com.ht117.htui.loading.CircleCountDownView;
import com.ht117.htui.loading.CountDownView;
import com.ht117.htui.loading.NumberLoadingView;

import java.util.Random;

public class WelcomeActivity extends AppCompatActivity implements SeekBar.OnSeekBarChangeListener, View.OnClickListener {

    private SeekBar seekbar;
    private CountDownView countDownView;
    private CircleCountDownView circleCountDownView;
    private NumberLoadingView numberLoadingView;
    private Handler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        findViewById(R.id.btnPlay).setOnClickListener(this);

        countDownView = (CountDownView) findViewById(R.id.countDownView);
        circleCountDownView = (CircleCountDownView) findViewById(R.id.circleCountDownView);
        numberLoadingView = (NumberLoadingView) findViewById(R.id.numberLoadingView);

        int duration = circleCountDownView.getDuration() / Utils.SECOND;
        ((TextView)findViewById(R.id.tvDuration)).setText(String.format("Duration: %ds", duration));

        seekbar = (SeekBar)findViewById(R.id.seekDuration);
        seekbar.setOnSeekBarChangeListener(this);
        seekbar.setProgress(duration);

        handler = new Handler();
        handler.postDelayed(randomProgress, Utils.SECOND / 4);
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        ((TextView)findViewById(R.id.tvDuration)).setText(String.format("Duration: %ds", seekBar.getProgress()));
        countDownView.setDuration(seekbar.getProgress());
        circleCountDownView.setDuration(seekbar.getProgress());
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {
    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {
        countDownView.setDuration(seekbar.getProgress());
        circleCountDownView.setDuration(seekbar.getProgress());
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btnPlay:
                countDownView.start();
                circleCountDownView.start();
                numberLoadingView.reset();
                handler.postDelayed(randomProgress, Utils.SECOND / 4);
                break;
        }
    }

    final Runnable randomProgress = new Runnable() {
        @Override
        public void run() {
            int prog = numberLoadingView.getProgress();
            if (prog < Utils.PERCENT) {
                Random rnd = new Random();
                int rndVal = rnd.nextInt(10);
                numberLoadingView.setProgress(prog + rndVal);
            }
            if (numberLoadingView.getProgress() < Utils.PERCENT) {
                handler.postDelayed(randomProgress, Utils.SECOND / 4);
            }
        }
    };
}

File: app/src/main/java/com/ht117/demo/WelcomeActivity.kt
package com.ht117.demo

import android.app.Activity
import android.os.Bundle
import android.os.Handler
import android.widget.SeekBar
import com.ht117.htui.Utils
import kotlinx.android.synthetic.main.activity_welcome.*
import java.util.*

class WelcomeActivity : Activity(), SeekBar.OnSeekBarChangeListener {

    private var handler: Handler? = null

    private val randomProgress: Runnable = object : Runnable {
        override fun run() {
            val prog = numberLoadingView!!.getProgress()
            if (prog < Utils.PERCENT) {
                val rnd = Random()
                val rndVal = rnd.nextInt(10)
                numberLoadingView!!.setProgress(prog + rndVal)
            }
            if (numberLoadingView!!.getProgress() < Utils.PERCENT) {
                handler!!.postDelayed(this, (Utils.SECOND / 4).toLong())
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        btnPlay.setOnClickListener {
            clickPlay()
        }

        val duration = circleCountDownView!!.getDuration() / Utils.SECOND
        tvDuration.text = String.format("Duration: %ds", duration)

        seekDuration.setOnSeekBarChangeListener(this)
        seekDuration.progress = duration

        handler = Handler()
        handler!!.postDelayed(randomProgress, (Utils.SECOND / 4).toLong())
    }

    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
        tvDuration.text = String.format("Duration: %ds", seekBar.progress)
        countDownView!!.setDuration(seekDuration!!.progress)
        circleCountDownView!!.setDuration(seekDuration!!.progress)
    }

    override fun onStartTrackingTouch(seekBar: SeekBar) {}

    override fun onStopTrackingTouch(seekBar: SeekBar) {
        countDownView!!.setDuration(seekBar.progress)
        circleCountDownView!!.setDuration(seekBar.progress)
    }

    private fun clickPlay() {
        countDownView!!.start()
        circleCountDownView!!.start()
        numberLoadingView!!.reset()
        handler!!.postDelayed(randomProgress, (Utils.SECOND / 4).toLong())
    }
}
